### ISSUE TITLE
Error creating Encrypted volume on USB-Drive

### ISSUE THREAD

**Issue by EpsiPepsi** [hardcore-sushi/DroidFS#330]
When I try to create a encrypted volume on a exfat USB-Drive I get the error "DroidFS has no write access to this path. Please try another location". But when i create a hidden volume with DroidFS it works, also i'm able to create a CryFS volume on the USB using my PC. 
The error is the same no matter if i try to create the volume on the root or a sub-directory on my drive.
logcat below

![Screenshot_20241231-152008_DroidFS](https://github.com/user-attachments/assets/51f520d9-ad18-43a4-a40a-65ebd0c5d505)

[DroidFS_2024-12-31_15_11_24.log](https://github.com/user-attachments/files/18282022/DroidFS_2024-12-31_15_11_24.log)

OS: /e/ OS on android 13

**Comment by hardcore-sushi** [comment]
I think it's due to the path resolving process. What path does DroidFS show after picking a directory?

If it starts with `/storage/F777-F535`, try to manually replace that with `/mnt/media_rw//F777-F535`.

I built a test APK implementing a potential fix. Could you try it?

https://upload.disroot.org/r/QRrHGUXg#s0JlpKUiMxOvpub9BTBSklQim2pOgOPSz4Mu4wwg8g4=

<details>
<summary>PGP signature</summary>

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA256

SHA256: ad57d113ea5413837b11e445cfc8a0f03b92e75916f7e89419502b04f73ef613
-----BEGIN PGP SIGNATURE-----

iHUEARYIAB0WIQS2Tv6GzuHQVPCCFxGv44Q0SkXhOgUCZ0s6xgAKCRCv44Q0SkXh
OkxVAP9ABeB2SJBFcxCg3nSntgykdHTRHjOgaiJYFZCQYEx7YwD+P6T9uj53X271
MtZ9ohdyRdbZRqRnWKV+r4WKQjxgIQ8=
=hKOM
-----END PGP SIGNATURE-----
```
</details>

**Comment by EpsiPepsi** [comment]
You are correct about the path. It shows /storage/7F77-9C34 and it worked after replacing it with /mnt/media_rw/7F77-9C34 and using a subfolder called Test.

But using the patched apk results in the same write error. 

Logcat from the patched apk:
[DroidFS_2025-01-02_06_02_28.log](https://github.com/user-attachments/files/18287968/DroidFS_2025-01-02_06_02_28.log)

Logcat from working normal apk:
[DroidFS_2025-01-02_06_13_23.log](https://github.com/user-attachments/files/18288007/DroidFS_2025-01-02_06_13_23.log)

**Comment by hardcore-sushi** [comment]
OK so what path does the patched APK show?

**Comment by EpsiPepsi** [comment]
It showed the correct format with `/mnt/media_rw/7F77-9C34/Test`

I tested again before responding and it does seem to work now! But i don't exactly know why it suddenly works. It's probably because i missed some files when deleting folders on my USB-Drive when testing the patched apk.

**Comment by hardcore-sushi** [comment]
I pushed the fix: 4d6b043a8ab8436e3547d2c9d81448ecb61245d5 It will be included in the next DroidFS release. Meanwhile, let me know if you encounter the issue again with the patched version.

**Comment by n60b** [comment]
for me, Android 13, work:
"If it starts with /storage/F777-F535, try to manually replace that with /mnt/media_rw//F777-F535"
replacement F777-F535 with correct ID...

**Comment by TheKeyLimePie** [comment]
I'm trying to open a Gocryptfs storage on my cloud storage (Seafile) and also get this r/w permission error.

Starting with the + button, im browsing to my cloud folder using the system's file browser which enters the following path to DroidFS' input field:
`/storage/my.seafile.com (user@example.com)`.

- First thing to note is that I've selected the Seafile app as the root data source and not the usual internal storage.
- Secondly, even though I've navigated to a specific directory in the file browser selection, the filled in input seems to end up at the root level of the storage; I would expect something like `/storage/my.seafile.com (user@example.com)/my/subfolder` instead of the path mentioned earlier.
- In `my/subfolder`, the files `gocryptfs.diriv` and `gocryptfs.conf` are listed (together with the encrypted files) when browse-selecting the folder

I tried to replace `/storage/` with `/mnt/media_rw/` as suggested in this thread but in my case it didn't solve the permission problem.

Snippet from the log file:
```
07-22 12:20:50.708 30848 30848 D PATH RESOLVER: Volume Id: my.seafile.com (user@example.com)
07-22 12:20:50.709 30848 30848 E ardcore.droidfs: hiddenapi: Accessing hidden method Landroid/os/storage/StorageVolume;->getPath()Ljava/lang/String; (runtime_flags=0, domain=platform, api=max-target-q,test-api) from Lsushi/hardcore/droidfs/util/Wiper; (domain=app) using reflection: denied
07-22 12:20:50.709 30848 30848 W ardcore.droidfs: hiddenapi: If this is a platform test consider enabling VMRuntime.ALLOW_TEST_API_ACCESS change id for this package.
07-22 12:20:50.709 30848 30848 D PATH RESOLVER: Volume Path: null
07-22 12:20:50.710 30848 30848 D PATH RESOLVER: External dir: /storage/emulated/0/Android/data/sushi.hardcore.droidfs/files
07-22 12:20:50.710 30848 30848 D PATH RESOLVER: getExternalFilesDirs failed
```

**Comment by hardcore-sushi** [comment]
@TheKeyLimePie Please see the [README.md](https://github.com/hardcore-sushi/DroidFS#limitations), [#6](https://github.com/hardcore-sushi/DroidFS/issues/6), [#25](https://github.com/hardcore-sushi/DroidFS/issues/25), and [#225](https://github.com/hardcore-sushi/DroidFS/issues/225).

DroidFS does its best to extract a file path from the URI returned by the system, but will inevitably fail if you select a cloud storage location. Can you post the full logcat so we can possibly blacklist the URI and display a more appropriate error message?
